src/Formats/FormatFactory.h

```diff
@@ -250,6 +250,8 @@ class FormatFactory final : private boost::noncopyable
     bool checkIfFormatHasAnySchemaReader(const String & name) const;
     bool checkIfOutputFormatPrefersLargeBlocks(const String & name) const;
 
+    bool checkParallelizeOutputAfterReading(const String & name, ContextPtr context) const;
```

Comment:
Logically, `FormatFactory` doesn't look like a good place for this function. It can be a standalone function. I'm not familiar with Formats code, so have no suggestion regarding it.

Reply:
Seems similar to `supportsSubsetOfColumns()`, `supportsSubcolumns()`, and `prefersLargeBlocks()`, which are all here. I don't think it's important where to put this. It would probably be good to reorganize all of FormatFactory and IFormat stuff at some point, but I don't have good specific ideas for that yet, and it seems low priority. E.g. maybe formats should be able to make use of the Processor graph for parallelizing parsing/formatting instead of rolling their own thread pools. Maybe IStorage capabilities flags should be all in a struct instead of a bunch of virtual methods, and format would just expose that. Idk.